python/cudf/cudf/utils/utils.py

```diff
@@ -394,3 +394,9 @@ def pa_mask_buffer_to_mask(mask_buf, size):
         dbuf.copy_from_host(np.asarray(mask_buf).view("u1"))
         return Buffer(dbuf)
     return Buffer(mask_buf)
+
+def isnat(val):
+    if not isinstance(val, (np.datetime64, np.timedelta64)):
+        return False
```

Comment:
If `val` is a string of `NaT` this will return `False`. Is this intended? Note numpy throws in this situation:
```python-traceback
In [50]: np.isnat('NaT')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-50-b9fab4fd69fc> in <module>
----> 1 np.isnat('NaT')

TypeError: ufunc 'isnat' is only defined for datetime and timedelta.
```

Reply:
This is mostly temporary just to get the logic straight - but really what it comes down to is do we want to support this kind of API call:

`cudf_obj.fillna('NaT')`

In which case we have to plumb this to the same codepath as

`cudf_obj.fillna(np.datetime64('NaT'))`

Currently we DO support this kind of API as part of current state cuDF

https://github.com/rapidsai/cudf/blob/branch-0.17/python/cudf/cudf/core/column/datetime.py#L117

although this can easily be changed.